share/smack/lib/smack.c

```diff
@@ -2228,19 +2228,95 @@ void __SMACK_init_func_memory_model(void) {
 }
 
 
-#if MEMORY_SAFETY || SIGNED_INTEGER_OVERFLOW_CHECK
+#if STRINGS_ENABLED || (MEMORY_SAFETY || SIGNED_INTEGER_OVERFLOW_CHECK)
```

Comment:
This compiler directive should not exist. String functions should be extracted to a separate file `share/smack/lib/string.c` which is conditionally included by the Smack scripts.